extensions/notebook/src/test/book/book.test.ts

```diff
@@ -449,4 +449,110 @@ describe('BookTreeViewProviderTests', function () {
+			let dataFolderPath = path.join(bookFolderPath, '_data');
+			let contentFolderPath = path.join(bookFolderPath, 'content');
+			let configFile = path.join(bookFolderPath, '_config.yml');
+			tableOfContentsFile = path.join(dataFolderPath, 'toc.yml');
+			let bookNotebookFile = path.join(contentFolderPath, 'notebook1.ipynb');
+			notebookFolderPath = path.join(rootFolderPath, 'NotebookTestData');
+			standaloneNotebookTitle = 'notebook2';
+			standaloneNotebookFile = path.join(notebookFolderPath, `${standaloneNotebookTitle}.ipynb`);
+			await fs.mkdir(rootFolderPath);
+			await fs.mkdir(bookFolderPath);
+			await fs.mkdir(dataFolderPath);
+			await fs.mkdir(contentFolderPath);
+			await fs.mkdir(notebookFolderPath);
+			bookTitle = 'Test Book';
+			await fs.writeFile(configFile, `title: ${bookTitle}`);
+			await fs.writeFile(tableOfContentsFile, '- title: Notebook1\n  url: /notebook1');
+			await fs.writeFile(bookNotebookFile, '');
+			await fs.writeFile(standaloneNotebookFile, '');
+
+			const mockExtensionContext = new MockExtensionContext();
+			appContext = new AppContext(mockExtensionContext, new ApiWrapper());
+			bookTreeViewProvider = new BookTreeViewProvider(appContext.apiWrapper, [], mockExtensionContext, false, 'bookTreeView');
+			let errorCase = new Promise((resolve, reject) => setTimeout(() => resolve(), 5000));
+			await Promise.race([bookTreeViewProvider.initialized, errorCase.then(() => { throw new Error('BookTreeViewProvider did not initialize in time'); })]);
+			appContext = new AppContext(undefined, new ApiWrapper());
+		});
+
+		it('should include books and notebooks when opening parent folder', async () => {
+			await bookTreeViewProvider.loadNotebooksInFolder(rootFolderPath);
+			should(bookTreeViewProvider.books.length).equal(2, 'Should have loaded a book and a notebook');
+
+			validateIsBook(bookTreeViewProvider.books[0]);
+			validateIsNotebook(bookTreeViewProvider.books[1]);
+		});
+
+		it('should include only books when opening books folder', async () => {
+			await bookTreeViewProvider.loadNotebooksInFolder(bookFolderPath);
+			should(bookTreeViewProvider.books.length).equal(1, 'Should have loaded only one book');
+
+			validateIsBook(bookTreeViewProvider.books[0]);
+		});
+
+		it('should include only notebooks when opening notebooks folder', async () => {
+			await bookTreeViewProvider.loadNotebooksInFolder(notebookFolderPath);
+			should(bookTreeViewProvider.books.length).equal(1, 'Should have loaded only one notebook');
+
+			validateIsNotebook(bookTreeViewProvider.books[0]);
+		});
+
+		this.afterEach(async function (): Promise<void> {
+			let bookItems = await bookTreeViewProvider.getChildren();
+			await Promise.all(bookItems.map(bookItem => bookTreeViewProvider.closeBook(bookItem)));
+		});
+
+		this.afterAll(async function (): Promise<void> {
+			if (await exists(rootFolderPath)) {
+				await promisify(rimraf)(rootFolderPath);
+			}
+		});
+
+		let validateIsBook = (book: BookModel) => {
```

Comment:
These seem more generically useful - suggest putting them in a general notebook test helper class

Reply:
The asserts are very specific to this test class. Converting it to a generic method would require a ton of arguments.

Reply:
Which still seems like a good thing to have. Wouldn't it basically be "compare two BookModels to see if they're equivalent"? You shouldn't need a ton of arguments - just the two I would thing.